trin-state/src/utils.rs

```diff
@@ -0,0 +1,116 @@
+use num::bigint::{BigInt, Sign};
+use num::Signed;
+
+// 2 ** 256
+const MODULO: [u8; 78] = [
+    49, 49, 53, 55, 57, 50, 48, 56, 57, 50, 51, 55, 51, 49, 54, 49, 57, 53, 52, 50, 51, 53, 55, 48,
+    57, 56, 53, 48, 48, 56, 54, 56, 55, 57, 48, 55, 56, 53, 51, 50, 54, 57, 57, 56, 52, 54, 54, 53,
+    54, 52, 48, 53, 54, 52, 48, 51, 57, 52, 53, 55, 53, 56, 52, 48, 48, 55, 57, 49, 51, 49, 50, 57,
+    54, 51, 57, 57, 51, 54,
+];
+
+// 2 ** 255
+const MID: [u8; 77] = [
+    53, 55, 56, 57, 54, 48, 52, 52, 54, 49, 56, 54, 53, 56, 48, 57, 55, 55, 49, 49, 55, 56, 53, 52,
+    57, 50, 53, 48, 52, 51, 52, 51, 57, 53, 51, 57, 50, 54, 54, 51, 52, 57, 57, 50, 51, 51, 50, 56,
+    50, 48, 50, 56, 50, 48, 49, 57, 55, 50, 56, 55, 57, 50, 48, 48, 51, 57, 53, 54, 53, 54, 52, 56,
+    49, 57, 57, 54, 56,
+];
+
+// distance function as defined at...
+// https://notes.ethereum.org/h58LZcqqRRuarxx4etOnGQ#Storage-Layout
+// todo: measure performance & optimize if necessary
+pub fn distance(node_id: BigInt, content_id: BigInt) -> BigInt {
+    let modulo = BigInt::parse_bytes(&MODULO, 10).unwrap();
+    let mid = BigInt::parse_bytes(&MID, 10).unwrap();
```

Comment:
There's probably some low-hanging-fruit to do this once, statically. But I'm not worried about that at the moment.